lib/Parse/Lexer.cpp

```diff
@@ -1905,7 +1905,7 @@ void Lexer::lexStringLiteral(unsigned CustomDelimiterLen) {
         wasErroneous = true;
 
         continue;
-        return formToken(tok::unknown, TokStart)
+        return formToken(tok::unknown, TokStart);
```

Comment:
I think we've run into a misunderstanding here. If you look at the PR diff for this function, the only changes we should have done so far are

* the `if (!isMultilineString) {` branch
* `|| CurPtr == BufferEnd`
* diagnose `string_interpolation_unclosed` and `opening_paren` inside the `else if`
* Your comment

Everything else, including the way the code manipulates control of the program with `return` and `continue`, stays the same.

Reply:
(Realized this btw) https://github.com/apple/swift/pull/58882/commits/caeffb91812bdc91314a00df8d767280d79cda0b

Reply:
You sure? We are returning in the branch with the condition, and continuing in the fallback branch. It used to be the other way around.

Reply:
Okay should actually now be correct with https://github.com/apple/swift/pull/58882/commits/1616a5c86523f51f394f88c75dec5c59c60abb9f